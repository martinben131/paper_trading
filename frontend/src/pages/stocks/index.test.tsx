import React from "react";
import { shallow } from "enzyme";
import Stocks from "./index";

jest.mock("../../store", () => ({
  useProfile: () => [
    {
      watchlist: [],
    },
  ],
  useStocks: () => [
    {
      haveStocksBeenFetched: true,
      data: [
        {
          code: "AAPL",
          name: "Apple Inc.",
          priceCurr: 123.45,
          priceChange: 1.26,
        },
      ],
    },
  ],
}));

jest.mock("react-router-dom", () => ({
  useHistory: () => ({}),
}));

test("Renders stocks page correctly", () => {
  const component = shallow(<Stocks />);
  expect(component).toMatchSnapshot();
});

test("Page contains Apple Inc", () => {
  const component = shallow(<Stocks />);
  expect(component.text()).toContain("AAPLApple Inc.$123.45+1.26%");
});
