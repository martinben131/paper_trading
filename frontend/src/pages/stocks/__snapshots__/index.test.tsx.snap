// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders stocks page correctly 1`] = `
<Fragment>
  <Blueprint3.InputGroup
    large={true}
    onChange={[Function]}
    placeholder="Enter stock code..."
    type="search"
  />
  <table
    className="stockTable bp3-html-table bp3-html-table-bordered bp3-html-table-condensed bp3-html-table-striped bp3-interactive sortable "
    id="StocksTable"
  >
    <thead>
      <tr>
        <th
          onClick={[Function]}
        >
          Stock Code 
        </th>
        <th
          onClick={[Function]}
        >
          Stock Name 
        </th>
        <th
          onClick={[Function]}
        >
          Price 
        </th>
        <th
          onClick={[Function]}
        >
          Day 
        </th>
        <th>
          Watchlist
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="AAPL"
      >
        <td
          onClick={[Function]}
        >
          <b>
            AAPL
          </b>
        </td>
        <td
          onClick={[Function]}
        >
          Apple Inc.
        </td>
        <td
          onClick={[Function]}
        >
          $
          123.45
        </td>
        <td
          onClick={[Function]}
        >
          <b
            className="up"
          >
            +1.26%
          </b>
        </td>
        <td>
          <Blueprint3.Button
            fill={true}
            icon="eye-off"
            intent="warning"
            minimal={true}
            onClick={[Function]}
          />
        </td>
      </tr>
    </tbody>
  </table>
</Fragment>
`;
